hera_cal/smooth_cal.py

```diff
@@ -171,247 +133,128 @@ def time_filter(gains, wgts, times, filter_scale = 120.0, nMirrors = 0):
+        one-to-one in time. This function sets up a time grid that spans the whole day with dt = integration time. 
+        Gains and flags are assigned to the nearest gridpoint using np.searchsorted. It is assumed that:
+        1) Sorting calfits_list and flags_npz_list string will put them in chronological order
+        2) All calfits and npzs have the same frequencies
+        3) The integration time does not change between files and integrations in a file are contiguous in time
+
         Arguments:
-            cal: UVCal object or path to calfits file that is to be smoothed
-            prev_cal: UVCal object, path to calfits file, or a list of either in chronological order,
-                that comes before the calibration to be smoothed.
-            next_cal: UVCal object, path to calfits file, or a list of either in chronological order,
-                that comes after the calibration to be smoothed.
+            calfits_list: list of string paths to calfits files containing calibration solutions and flags
+            flags_npz_list: list of string paths to npz files containing flags as a function of baseline, times
+                and frequency. Must have all baselines for all times. Flags on baselines are broadcast to both 
+                antennas involved, unless either antenna is completely flagged for all times and frequencies.
         '''
-        assert(isinstance(cal, (str, UVCal)))
-        self.input_cal = cal
-        self.gains, self.flags, self.quals, _, _, self.freqs, self.times, self.pols =  io.load_cal(cal, return_meta=True)
-        self.nFreq = len(self.freqs)
-        self.nInt =len(self.times)
-        self.tInt = np.median(np.diff(self.times))*24.0*60.0*60.0
-        self.has_cal = True
+        # load calibration files
+        self.cals = sorted(calfits_list)
+        self.gains, self.cal_flags, self.cal_freqs, self.cal_times = odict(), odict(), odict(), odict()
+        for cal in self.cals:
+            (self.gains[cal], self.cal_flags[cal], _, _, _, self.cal_freqs[cal], 
+                 self.cal_times[cal], _) = io.load_cal(cal, return_meta=True)
+            
+        # load flags files
+        self.npzs = sorted(flags_npz_list)
+        if len(self.npzs) > 0:
+            self.npz_flags, self.npz_freqs, self.npz_times = odict(), odict(), odict()
+            for cal, npz in zip(self.cals, self.npzs):
+                self.npz_flags[cal] = synthesize_ant_flags(io.load_npz_flags(npz))
+                npz_dict = np.load(npz)
+                self.npz_freqs[cal] = np.unique(npz_dict['freq_array'])
+                self.npz_times[cal] = np.unique(npz_dict['time_array'])
+                    
+        # set up time grid
+        all_file_times = np.array(self.cal_times.values()).ravel()
+        self.dt = np.median(np.diff(all_file_times))
+        self.time_grid = np.arange(all_file_times[0], all_file_times[-1]+self.dt, self.dt)
```

Comment:
may be possible that all_file_times[-1]+self.dt has a round-off error that makes it epsilon larger, so that it becomes the endpoint.  Suggest adding self.dt/2 to ensure this doesn't happen.

Reply:
Fixed.